From 66b5770a2ad2955a41c4ff93fb009a23d7f11f4b Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Thu, 3 Aug 2017 18:52:44 +0300
Subject: [PATCH 0576/1051] irqchip/gicv3-its: update cache flushing print
 location

Update cache flushing print location, depends
on ITS_FLAGS_CMDQ_NEEDS_FLUSHING

Change-Id: Ifed206e3b57fa6ae75457359555074cd23a1db07
Signed-off-by: Hanna Hawa <hannah@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/irqchip/irq-gic-v3-its.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/irqchip/irq-gic-v3-its.c b/drivers/irqchip/irq-gic-v3-its.c
index 24800620a7fe..e86ab30f6137 100644
--- a/drivers/irqchip/irq-gic-v3-its.c
+++ b/drivers/irqchip/irq-gic-v3-its.c
@@ -3454,7 +3454,6 @@ static int __init its_probe_one(struct resource *res,
 			baser |= GITS_CBASER_nC;
 			gits_write_cbaser(baser, its->base + GITS_CBASER);
 		}
-		pr_info("ITS: using cache flushing for cmd queue\n");
 		its->flags |= ITS_FLAGS_CMDQ_NEEDS_FLUSHING;
 	}
 
@@ -3475,6 +3474,9 @@ static int __init its_probe_one(struct resource *res,
 		writeq_relaxed(baser, its->base + GITS_CBASER);
 	}
 
+	if (its->flags & ITS_FLAGS_CMDQ_NEEDS_FLUSHING)
+		pr_info("ITS: using cache flushing for cmd queue\n");
+
 	gits_write_cwriter(0, its->base + GITS_CWRITER);
 	ctlr = readl_relaxed(its->base + GITS_CTLR);
 	ctlr |= GITS_CTLR_ENABLE;
-- 
2.17.1

